avro_schema/utils.lua

```diff
@@ -7,6 +7,25 @@ local function table_contains(t, xval)
     return false
 end
 
+-- This function copies fields from one table to the other deeply.
+-- In case `fields` param passed, only fields are copied.
+--
+-- From is an array of field names.
+local function copy_fields(from, to, fields)
+    if fields then
+        for _,field in ipairs(fields) do
+            if from[field] ~= nil then
+                to[field] = table.deepcopy(from[field])
```

Comment:
There are two things to consider here:

1. Here we use deepcopy, but in case when no fields passed we use shallow copy (by the way it can be just table.copy()).
2. table.copy and table.deepcopy appears somewhere in 1.7. We don't intend to support avro-schema for 1.6, so it is good to use these functions (but we need to consider possible differences, e.g. type-check (is a table) in the our built-in functions).